x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/CcrSettings.java

```diff
@@ -32,6 +32,12 @@ private CcrSettings() {
     public static final Setting<Boolean> CCR_FOLLOWING_INDEX_SETTING =
             Setting.boolSetting("index.xpack.ccr.following_index", false, Setting.Property.IndexScope);
 
+    /**
+     * An internal setting used to keep track to what leader index a follow index associated. Do not use externally!
```

Comment:
I think that we need a different mechanism here. Either we need to invent the notion of a private setting, or as I prefer, these should be considered custom index metadata. While we have partially deprecated and neutered custom index metadata, I think that we should bring it back. We also have a similar need in index lifecycle management. As @colings86, @talevy, and me are all out next week, I will pick up a thread when I return from vacation to resuscitate custom index metadata and then we can use it here and in index lifecycle management.

Reply:
Thanks @jasontedor for sharing your thoughts on this setting. When I added the setting, it felt wrong to me that this was a public setting, hence the comment. I think your idea to use custom index metadata makes sense here.